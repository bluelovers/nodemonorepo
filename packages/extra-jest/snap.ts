import * as yaml from 'js-yaml'

export function mkfn<X> (fn: (x: X) => string): (x: X) => () => void {
  return (x: X) => () => {
    expect(fn(x)).toMatchSnapshot()
  }
}

export const unsafe = mkfn(x => yaml.dump(x))
export const safe = mkfn(x => yaml.safeDump(x))
