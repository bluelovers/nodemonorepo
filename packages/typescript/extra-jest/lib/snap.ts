import * as yaml from 'js-yaml'

export type DumpOptions = yaml.DumpOptions
export type DumpFunction<X> = (x: X, o?: DumpOptions) => string
export type TestFunction<X> = (x: X, o?: DumpOptions) => () => void

export function mkfn<X> (fn: DumpFunction<X>, options: DumpOptions = {}): TestFunction<X> {
  return (x: X, extraOptions: DumpOptions = {}) => () => {
    expect('\n' + fn(x, { ...options, ...extraOptions }) + '\n').toMatchSnapshot()
  }
}

const defaultOptions: DumpOptions = {
  skipInvalid: true,
  sortKeys: true,
  noRefs: false,
  noCompatMode: true
}

const unsafeDump: DumpFunction<any> = (x, o) => yaml.dump(x, o)
const safeDump: DumpFunction<any> = (x, o) => yaml.safeDump(x, o)

export const unsafe = mkfn(unsafeDump, defaultOptions)
export const safe = mkfn(safeDump, defaultOptions)
export const pureUnsafe = mkfn(unsafeDump)
export const pureSafe = mkfn(safeDump)
export const noRefs = mkfn(unsafeDump, { noRefs: true })

export default unsafe
