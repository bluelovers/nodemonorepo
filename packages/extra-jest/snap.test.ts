import * as snap from './snap'

const unsafePattern = {
  primaries: {
    numbers: {
      integer: 123,
      float: 45.6,
      infinity: Infinity,
      zero: 0,
      nan: NaN
    },
    string: 'Hello, World!!',
    nothingness: {
      null: null,
      undefined: undefined
    }
  },
  references: {
    object: {
      abc: 123,
      def: 456
    },
    array: [
      [...'abcdef'],
      [123, 456, 789]
    ],
    function: () => 'Hello, World!!'
  }
}

const safePattern = {
  primaries: {
    numbers: {
      integer: 123,
      float: 45.6,
      infinity: Infinity,
      zero: 0,
      nan: NaN
    },
    string: 'Hello, World!!',
    nothingness: {
      null: null
    }
  },
  references: {
    object: {
      abc: 123,
      def: 456
    },
    array: [
      [...'abcdef'],
      [123, 456, 789]
    ]
  }
}

describe('snap.unsafe', () => {
  it('works', snap.unsafe(unsafePattern))
})

describe('snap.safe', () => {
  it('works', snap.safe(safePattern))
})
