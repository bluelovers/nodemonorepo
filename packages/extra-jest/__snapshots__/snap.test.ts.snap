// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap.safe works 1`] = `
"primaries:
  numbers:
    integer: 123
    float: 45.6
    infinity: .inf
    zero: 0
    nan: .nan
  string: 'Hello, World!!'
  nothingness:
    'null': null
references:
  object:
    abc: 123
    def: 456
  array:
    - - a
      - b
      - c
      - d
      - e
      - f
    - - 123
      - 456
      - 789
"
`;

exports[`snap.unsafe works 1`] = `
"primaries:
  numbers:
    integer: 123
    float: 45.6
    infinity: .inf
    zero: 0
    nan: .nan
  string: 'Hello, World!!'
  nothingness:
    'null': null
    undefined: !<tag:yaml.org,2002:js/undefined> ''
references:
  object:
    abc: 123
    def: 456
  array:
    - - a
      - b
      - c
      - d
      - e
      - f
    - - 123
      - 456
      - 789
  function: !<tag:yaml.org,2002:js/function> '() => ''Hello, World!!'''
"
`;
